index.js

```diff
@@ -88,19 +88,213 @@ module.exports = {
+                            'q-applicant-confirmation-method': {
+                                const: 'none'
+                            }
+                        },
+                        required: ['q-applicant-confirmation-method']
+                    },
+                    then: {
+                        additionalProperties: false,
+                        properties: {
+                            'q-applicant-confirmation-method': {
+                                const: 'none'
+                            }
+                        },
+                        required: ['q-applicant-confirmation-method']
+                    }
+                }
+            },
+            errorMessage: {
+                required: {
+                    'q-applicant-confirmation-method':
+                        'Select how you want to get your confirmation message'
+                }
+            },
+            examples: [
+                {
+                    'q-applicant-confirmation-method': 'none'
+                },
+                {
+                    'q-applicant-confirmation-method': 'email',
+                    'q-applicant-enter-your-email-address': 'foo@bar.com'
+                },
+                {
+                    'q-applicant-confirmation-method': 'sms',
+                    'q-applicant-enter-your-telephone-number': '07701 234567'
+                }
+            ],
+            invalidExamples: [
+                {
+                    'q-applicant-confirmation-method': 'none',
+                    'q-applicant-enter-your-email-address': 'foo@bar.com'
+                },
+                {
+                    'q-applicant-confirmation-method': 'none',
+                    'q-applicant-enter-your-telephone-number': '07701 234567'
+                },
+                {
+                    'q-applicant-confirmation-method': 'email'
+                },
+                {
+                    'q-applicant-confirmation-method': 'sms'
+                },
+                {
+                    'q-applicant-confirmation-method': 'email',
+                    'q-applicant-enter-your-telephone-number': '07701 234567'
+                },
+                {
+                    'q-applicant-confirmation-method': 'sms',
+                    'q-applicant-enter-your-email-address': 'foo@bar.com'
+                },
+                {
+                    'q-applicant-confirmation-method': 'email',
+                    'q-applicant-enter-your-email-address': 'not an email address'
+                },
+                {
+                    'q-applicant-confirmation-method': 'sms',
+                    'q-applicant-enter-your-telephone-number': 'not a UK mobile phone number'
+                },
+                {
+                    'q-applicant-confirmation-method': 'sms',
+                    'q-applicant-enter-your-telephone-number': '0141 420 5000'
+                }
+            ]
+        },
```

Comment:
For consistency, can we test that incorrect data types also fail. E.g If we submit an integer to the schema it will the schema fail?